### ISSUE TITLE
dvportgroup names should be referenced by `key`

### ISSUE THREAD

**Issue by nward** [fog/fog-vsphere#92]
See http://projects.theforeman.org/issues/19623 for back story and a possible patch.

Happy to submit one myself if you'd like.

(I am not the author of the above link, but, have come across the same problem)

**Comment by jrgarcia** [comment]
@nward Feel free to submit a PR. I don't mind doing it, but it may be a while before I get a chance to look at that backstory and patch.

**Comment by wiad** [comment]
I am the OP of the bug report on theforemans bug tracker, submitting a PR of my solution to this 'problem'.

**Comment by wiad** [comment]
The bug report at theforeman is kind of sprawling, I will try to explain the problem in a more condensed form:
My problem was that I was getting different identifiers for my vmware dv portgroups in Foreman - when I created a machine the network was identified with one value (dvportgroup-XX) but when I tried to edit the machine validation failed since the dvportgroup was then identified with a different value. After a lot of investigating and checking with vmware, it is actually possible to have different identifiers (i.e. different key and ref attribute) for the same dvportgroup under certain circumstances - after a migration of vcenter for example.
